src/app/login-register/login-register.component.ts

```diff
@@ -2,20 +2,112 @@ import { Component, OnInit } from '@angular/core';
 import {MatIconModule} from '@angular/material/icon';
 import {MatFormFieldModule} from '@angular/material/form-field';
 import {MatCheckboxModule} from '@angular/material/checkbox';
+import {FormControl, Validators, ReactiveFormsModule} from '@angular/forms';
+import { AbstractControl, NG_VALIDATORS, ValidationErrors, Validator, ValidatorFn } from '@angular/forms';
+import {Router} from "@angular/router";
+import {UserRegisterService} from "./service/user-register.service";
 
 @Component({
   selector: 'app-login-register',
   templateUrl: './login-register.component.html',
   styleUrls: ['./login-register.component.scss']
 })
-export class LoginRegisterComponent implements OnInit {
+
+export class LoginRegisterComponent /*implements OnInit*/ {
 
   hidePassword = true;
   hideConfirmPassword = true;
+  email = new FormControl('', [Validators.required, Validators.email]);
+  password = new FormControl('', [Validators.minLength(8), Validators.required]);
+  confirmPassword = new FormControl('', [Validators.required, this.equalPassword('password')]);
+  acceptDSGVO = new FormControl(false, [Validators.requiredTrue]);
+  hasError = true;
+  registerSuccess = false;
+  userId: Number;
+
+  constructor(private userRegisterService: UserRegisterService, private router: Router) { }
+
+  getError() {
+    if(this.email.invalid || this.password.invalid /*|| this.confirmPassword.invalid*/ || this.acceptDSGVO.invalid) {
+      this.hasError = true;
+    } else {
+      this.hasError = false;
+    }
+    return this.hasError;
+  }
+
+  getErrorMessageMail() {
+    if(this.email.hasError('required')) {
+      return 'Please enter your mail address!';
+    } else if(this.email.hasError('email')) {
+      return 'It is not a valid email!';
+    } else {
+      return '';
+    }
+  }
+
+  getErrorMessagePassword() {
+    if(this.password.hasError('required')) {
+      return 'Please enter your password!';
+    } else if(this.password.hasError('minlength')) {
+      return 'The password is to short! (min length 8)';
+    } else {
+      return '';
+    }
+  }
+
+  getErrorMessageConfirmPassword() {
+    if(this.confirmPassword.hasError('required')) {
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/MyCookieBook/CookieBookFE/pullRequest?prid=7552947)
```suggestion
    if(this.confirmPassword.hasError("required")) {
```